src/main/kotlin/org/rust/lang/core/completion/RsCommonCompletionProvider.kt

```diff
@@ -281,6 +284,28 @@ private fun filterVisRestrictionPaths(
     }
 }
 
+/**
+ * Ignore items that are not traits inside trait refs:
+ * `impl /*here*/ for <type>`
+ * `fn foo<T: /*here*/>() {}
+ */
+private fun filterTraitRefPaths(
+    path: RsPath,
+    processor: RsResolveProcessor
+): RsResolveProcessor {
+    val parent = path.parent
+    return if (parent is RsTraitRef) {
+        createProcessor(processor.name) {
+            when (it.element) {
+                !is RsTraitItem -> false
```

Comment:
We should also complete modules since someone may want to type a qualified path in a trait ref

Reply:
Good point! Thanks.